### DISCUSSION TITLE
Multiple targets match implicit dependency

### DISCUSSION THREAD

**Discussion by dalton5** [firebase/flutterfire#7141] [Q&A] [answered]
Hi,

I have an issue with Firebase messaging

When I want to archive my ipa I have the issue:

warning: Multiple targets match implicit dependency for linker flags
    '-framework GoogleUtilities'. Consider adding an explicit dependency
    on the intended target to resolve this ambiguity. (in target
    'Runner' from project 'Runner')
        note: Target 'GoogleUtilities-54e75ca4' (in project 'Pods')
        note: Target 'GoogleUtilities-ac14cd06' (in project 'Pods')

My Pod file is as below:

What is wrong? How can I fix it?

                  `# Uncomment this line to define a global platform for your project
                  platform :ios, '11.0'
                  
                  # CocoaPods analytics sends network stats synchronously affecting flutter build latency.
                  ENV['COCOAPODS_DISABLE_STATS'] = 'true'
                  
                  project 'Runner', {
                    'Debug' => :debug,
                    'Profile' => :release,
                    'Release' => :release,
                  }
                  
                  def flutter_root
                    generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
                    unless File.exist?(generated_xcode_build_settings_path)
                      raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
                    end
                  
                    File.foreach(generated_xcode_build_settings_path) do |line|
                      matches = line.match(/FLUTTER_ROOT\=(.*)/)
                      return matches[1].strip if matches
                    end
                    raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
                  end
                  
                  require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)
                  
                  flutter_ios_podfile_setup
                  
                  target 'Runner' do
                    use_frameworks!
                    use_modular_headers!
                  
                    flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
                  end
                  
                  
                  
                  post_install do |installer|
                    installer.pods_project.targets.each do |target|
                      flutter_additional_ios_build_settings(target)
                      target.build_configurations.each do |config|
                        # Here are some configurations automatically generated by flutter
                  
                        # You can enable the permissions needed here. For example to enable camera
                        # permission, just remove the `#` character in front so it looks like this:
                        #
                        # ## dart: PermissionGroup.camera
                        # 'PERMISSION_CAMERA=1'
                        #
                        #  Preprocessor definitions can be found in: https://github.com/Baseflow/flutter-permission-handler/blob/master/permission_handler/ios/Classes/PermissionHandlerEnums.h
                        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
                          '$(inherited)',
                  
                          ## dart: PermissionGroup.calendar
                          # 'PERMISSION_EVENTS=1',
                  
                          ## dart: PermissionGroup.reminders
                          # 'PERMISSION_REMINDERS=1',
                  
                          ## dart: PermissionGroup.contacts
                          # 'PERMISSION_CONTACTS=1',
                  
                          ## dart: PermissionGroup.camera
                          'PERMISSION_CAMERA=1',
                  
                          ## dart: PermissionGroup.microphone
                          # 'PERMISSION_MICROPHONE=1',
                  
                          ## dart: PermissionGroup.speech
                          # 'PERMISSION_SPEECH_RECOGNIZER=1',
                  
                          ## dart: PermissionGroup.photos
                         'PERMISSION_PHOTOS=1',
                  
                          ## dart: [PermissionGroup.location, PermissionGroup.locationAlways, PermissionGroup.locationWhenInUse]
                          # 'PERMISSION_LOCATION=1',
                  
                          ## dart: PermissionGroup.notification
                          # 'PERMISSION_NOTIFICATIONS=1',
                  
                          ## dart: PermissionGroup.mediaLibrary
                          # 'PERMISSION_MEDIA_LIBRARY=1',
                  
                          ## dart: PermissionGroup.sensors
                          # 'PERMISSION_SENSORS=1',   
                  
                          ## dart: PermissionGroup.bluetooth
                          # 'PERMISSION_BLUETOOTH=1',
                  
                          ## dart: PermissionGroup.appTrackingTransparency
                          # 'PERMISSION_APP_TRACKING_TRANSPARENCY=1',
                  
                          ## dart: PermissionGroup.criticalAlerts
                          # 'PERMISSION_CRITICAL_ALERTS=1'
                        ]
                  
                      end
                    end
                  end
                  
                  
                  target 'ImageNotification' do
                    use_frameworks!
                    pod 'Firebase/Messaging'
                  end
                  `

**Comment by dalton5** [comment, marked as answer]
Ok I fix it.

I added pod 'GoogleUtilities' in both target in my pod file as below

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
  pod 'GoogleUtilities'
end

target 'ImageNotification' do
  use_frameworks!
  pod 'Firebase/Messaging'
  pod 'GoogleUtilities'
end

**Reply by salarazad** [reply]
thank you this finally worked for me.

**Comment by lisovyk** [comment]
I had a similar issue; my relevant podfile parts looked like this:

```
target 'Runner' do
  use_frameworks!
  use_modular_headers!

  pod 'GoogleTagManager', :modular_headers => true
  pod 'Firebase', :modular_headers => true
  pod 'FirebaseCore', :modular_headers => true
  pod 'FirebaseCoreInternal', :modular_headers => true
  pod 'GoogleUtilities', :modular_headers => true

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end

target 'ImageNotification' do
  use_frameworks!
  pod 'Firebase/Messaging'
end
```

Respectively, changed `ImageNotification` target to:

```
target 'ImageNotification' do
  use_frameworks!
  use_modular_headers!
  pod 'Firebase/Messaging', :modular_headers => true
  pod 'GoogleUtilities', :modular_headers => true
end
```
